### DISCUSSION TITLE
How can I make selections only applied to filtered results in an HoC?

### DISCUSSION THREAD

**Discussion by gandhis1** [dowjones/react-dropdown-tree-select#572] [Q&A]
So I have implemented an HoC for a number of other customizations that control what is selected and deselected when a node is checked. Another one that I am looking to do is to account for search/filter state. So if I filter a tree, and then a node is selected, I want only the visible items to be selected. However, as far as I can tell nothing about the search state (visible or not) is exposed on a `TreeNode` object, so I'm not sure whether this is even possible.

**Comment by mrchief** [comment]
You an only select the visible items but if those items happen to be parent nodes, they can in turn select the children. If you don't want this behavior, you can set `mode=hierarchical`. [docs](https://github.com/dowjones/react-dropdown-tree-select#hierarchical)

**Reply by gandhis1** [reply]
I understand that - my HoC already does a custom selection of the children (leaf nodes only). But I want to update that logic to select only visible leaf nodes if a filter is active. So fundamentally my question is, for me to be able to do this in the HoC, I would need some way to identify either (1) which nodes are visible or (2) what the search criteria is.

**Reply by mrchief** [reply]
Oh I see. I'm afraid we don't have a way to surface that right now. I guess we can create an event `onSearch` or `onFilter` and send a list of nodes that match. I'll be happy to accept a PR
